azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -34,16 +52,37 @@ class CGroupsTelemetry(object):
     _cgroup_metrics = {}
     _rlock = threading.RLock()
 
+    @staticmethod
+    def get_process_info_summary(process_id):
+        process_cmdline = DEFAULT_PROCESS_COMMANDLINE
+        process_name = DEFAULT_PROCESS_NAME
+
+        # The ProcessName and ProcessCommandLine can generate Exception if the file /proc/<pid>/{comm,cmdline} cease to
+        # exist; eg: the process can die, or finish. Which is why we need Default Names, in case we fail to fetch the
+        # details from those files.
+        try:
+            process_cmdline = ProcessInfo.get_proc_cmdline(process_id) if not None else DEFAULT_PROCESS_COMMANDLINE
+        except Exception as e:
+            logger.periodic_info(EVERY_SIX_HOURS, "[PERIODIC] {0}", ustr(e))
+
+        try:
+            process_name = ProcessInfo.get_proc_name(process_id) if not None else DEFAULT_PROCESS_NAME
+        except Exception as e:
+            logger.periodic_info(EVERY_SIX_HOURS, "[PERIODIC] {0}", ustr(e))
+
+        return process_id + DELIM + process_name + DELIM + process_cmdline
```

Comment:
Small comment, why not use "{0}".format() instead? That makes the string look cleaner and easy to debug

Reply:
Though it was simple enough and wanted to keep the structure explicit. I actually felt this was a little unclear to what the generated structure would be.
```
summarized_info = "{1} {0} {2} {0} {3}".format(DELIM, process_id, process_name, process_cmdline)
```

I can fix this in the later PRs if this is not clear enough.